app/main.py

```diff
@@ -1,25 +1,71 @@
 from abc import ABC
+from collections.abc import Callable
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if isinstance(value, int):
+            if self.min_amount <= value <= self.max_amount:
+                setattr(instance, self.protected_name, value)
+            else:
+                raise ValueError("You can't slide here :(")
+        else:
+            raise TypeError("Those are not real human parameters")
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
```

Comment:
The `SlideLimitationValidator` class should be abstract. Consider adding an abstract method that subclasses must implement to validate the visitor's attributes.